src/playwright/pages/base.page.ts

```diff
@@ -0,0 +1,103 @@
+
+type LanguageButton = 'Suomi' | 'Svenska' | 'English';
+type HeaderButton = LanguageButton;
+
+type AccessibilityStatementLink =
+  | 'Saavutettavuusseloste'
+  | 'Tillgänglighetsutlåtande'
+  | 'Accessibility Statement';
+type FooterLink = AccessibilityStatementLink;
+
+type SearchButton = 'Hae tapahtumia' | 'Sök evenemang' | 'Search events';
+type TargetGroupButton = 'Kohderyhmät' | 'Målgrupp' | 'Target groups';
+
+type ZeroToTwoYears = '0-2 vuotiaat' | '0-2 år' | '0-2 years';
+type TargetGroupOption = ZeroToTwoYears;
+type MainContentOption = TargetGroupOption;
+
+type MainContentTextBox = SearchButton;
+
+type MainContentButton = SearchButton | TargetGroupButton;
+
+export class BasePage {
+  protected readonly page: Page;
+  protected readonly h1: Locator;
+  protected readonly footer: Locator;
+  protected readonly header: Locator;
+  protected readonly mainContent: Locator;
+  protected readonly cookieConsentBox: Locator;
+  protected readonly cookieConsentApprove: Locator;
+
+  constructor(page: Page) {
+    if (this.constructor === BasePage) {
+      throw new Error(
+        'Abstract BasePage class cannot be instantiated directly'
+      );
+    }
+    this.page = page;
+    this.h1 = page.locator('h1');
+    this.header = page.getByRole('banner');
+    this.footer = page.getByRole('contentinfo');
+    this.mainContent = page.locator('#main-content');
+    this.cookieConsentBox = page.locator('#cookie-consent-content');
+    this.cookieConsentApprove = page.getByTestId(
+      'cookie-consent-approve-required-button'
+    );
+  }
+
+  async isReady() {
+    await this.page.waitForLoadState('networkidle', IN_20s); // Network can be very slow
+  }
+
+  async reload() {
+    return await this.page.reload();
+  }
+
+  async clickFooterLink(name: FooterLink) {
+    await this.footer.getByRole('link', { name }).first().click();
+  }
+
+  async clickHeaderButton(name: HeaderButton) {
+    await this.header.getByRole('button', { name }).first().click();
+  }
+
+  async clickSearchOption(name: MainContentOption) {
+    await this.mainContent.getByRole('option', { name }).first().click();
+  }
+
+  async fillSearchTextBox(name: MainContentTextBox, value: string) {
+    const searchTextBox = this.page.getByRole('textbox', { name });
+    await searchTextBox.click(); // First open the text box by clicking on it
+    await searchTextBox.fill(value);
+  }
+
+  async clickSearchButton(name: MainContentButton) {
+    await this.mainContent.getByRole('button', { name }).first().click();
+  }
+
+  async isCookieConsentBoxVisible(timeout?: number) {
+    await expect(this.cookieConsentBox).toBeVisible({ timeout });
+  }
+
+  async isCookieConsentBoxHidden(timeout?: number) {
+    await expect(this.cookieConsentBox).not.toBeVisible({ timeout });
+  }
+
+  async isCookieConsentBoxHiddenAfterMs(delayMs: number) {
+    await this.page.waitForTimeout(delayMs);
+    await this.isCookieConsentBoxHidden(0); // Instant check
+  }
```

Comment:
Using keyword arguments would be good practise, because I have no idea what this `isCookieConsentBoxHidden` argument with value `0` is. Timeouts are configured with keyword arguments also in testing-library and also in Playwright.

```typescript
type LocatorOptions = {
  timeout?: number;
  delayMs?: number;
};
```

and

```suggestion
  async isCookieConsentBoxVisible({
    timeout,
  }: Pick<LocatorOptions, 'timeout'>) {
    await expect(this.cookieConsentBox).toBeVisible({ timeout });
  }

  async isCookieConsentBoxHidden({ timeout, delayMs }: LocatorOptions) {
    if (delayMs) {
      await this.page.waitForTimeout(delayMs);
    }
    await expect(this.cookieConsentBox).not.toBeVisible({ timeout });
  }
```

Reply:
Made use of Timeout type more in 3a5abd212b5c7c344a004ae21739f6c0832e1eb6. Left the delayMs as it was as IMHO the function is clear as the `AfterMs` suffix tells what the parameter means.

Reply:
Used Timeout type more in 88a56d7c0fd58126bbbfac90e1decdc47130e441 and split the delaying into its own function `waitForMs(delayMs: number)`. Now this should be clear IMHO.